### ISSUE TITLE
Pooled addresses are removed in scale down process

### ISSUE THREAD

**Issue by kornys** [EnMasseProject/enmasse#2100]
**Describe the bug**
In progress of scale down of pooled broker statefull set, broker pods with active addresses are deleted.

**To Reproduce**
1. Create a lot of addresses to scale up pooled brokers (for example to 4 pods)
2. Delete **first** half of addresses and wait for scale down (for example to 2 pods)

**Expected behavior**
enmasse should keep broker pods with not deleted addresses 

```
oc get address many-addresses-standard.xxs-queue-2000 -o wide
NAME                                     ADDRESS          ADDRESSSPACE              TYPE      PLAN               READY     PHASE     AGE       STATUS
many-addresses-standard.xxs-queue-2000   xxs-queue-2000   many-addresses-standard   queue     pooled-xxs-queue   false     Active    2h        Address xxs-queue-2000 is missing active autoLink (active in dirs: [])
```

**Comment by lulf** [comment]
CC @calohmn

**Comment by lulf** [comment]
Fixed in #2101
